packages/@aws-cdk/aws-eks/lib/cluster-resource-handler/cluster.ts

```diff
@@ -198,6 +205,8 @@ export class ClusterResourceHandler extends ResourceHandler {
           Endpoint: cluster.endpoint,
           Arn: cluster.arn,
           CertificateAuthorityData: cluster.certificateAuthority?.data,
+          ClusterSecurityGroupId: cluster.resourcesVpcConfig?.clusterSecurityGroupId,
+          EncryptionConfigKeyArn: cluster.encryptionConfig?.shift()?.provider?.keyArn,
```

Comment:
This is not documented or mentioned in the PR title/description.

Reply:
Also, add a comment explaining why it is safe to return only the first item

Reply:
update the title and description with more info and relevant links

Reply:
> Also, add a comment explaining why it is safe to return only the first item

added comments there, but basically this array has "Maximum number of 1 item".